### ISSUE TITLE
Conditional build - is it possible?

### ISSUE THREAD

**Issue by gtamas** [aws/copilot-cli#2382]
I have a question about copilot manifests:

In the "environments" section I should be able to override any setting, right?

It seems to me, however, the "image.build.*" section cannot be overridden. Or should this work fine?

```
 build: 
    dockerfile: some/path/Dockerfile_test
    no_cache: true


environments:
  prod:
    build: 
      dockerfile: siome/other/Dockerfile_prod
      no_cache: true

```

I can't override anything under image.build. Tried "args" too, same result. But I need to copy certain resources into the docker container conditionally, depending on which copilot environment I'm deploying to. 

How do I solve this? Please help!

Thanks in advance

**Comment by gtamas** [comment]
Just to make it clear, this also doesn't work:

```
 build: 
    dockerfile: some/path/Dockerfile
    no_cache: true
    args:
      foo: 1


environments:
  prod:
    build: 
      dockerfile: siome/other/Dockerfile
      no_cache: true
      args:
        foo: 2
```

Foo is always 1, regardless of the environment value.

I need to build certain things conditionally, how do I do that with copilot? It seems to me that everything else can be overridden but image.build.*

Since args doesn't work, I can't solve it on the Docker level either.

What should I do??

**Comment by bvtujo** [comment]
Hey @gtamas, you're right that this is currently a bug. Overrides don't work for docker build, as we originally intended the environment overrides to be a Cloudformation-only construct. There's no reason, however, for us to limit that to the CF fields. We should allow overriding build args. 

In the meantime, can you build your images manually and push them to the ECR repo that Copilot creates? You can then use the use the `image.location` field to specify the right image per environment.

**Comment by efekarakus** [comment]
This is fixed in #2415 and will go out next release!

**Comment by efekarakus** [comment]
This is now out in v1.8.0 🥳 https://github.com/aws/copilot-cli/releases/tag/v1.8.0
